src/test_util.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+use std::convert::Infallible;
+
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::storage::StateKind;
+
+fn generate_key<T: StateKind>(lattice_id: &str) -> String {
+    format!("{}_{lattice_id}", T::KIND)
+}
+
+/// A [`Store`] implementation for use in testing
+#[derive(Default)]
+pub struct TestStore {
+    pub inner: tokio::sync::RwLock<HashMap<String, Vec<u8>>>,
+}
+
+#[async_trait::async_trait]
+impl crate::storage::Store for TestStore {
+    type Error = Infallible;
+
+    async fn get<T>(&self, lattice_id: &str, id: &str) -> Result<Option<T>, Self::Error>
+    where
+        T: DeserializeOwned + StateKind,
+    {
+        let key = generate_key::<T>(lattice_id);
+        let mut all: HashMap<String, T> = self
+            .inner
+            .read()
+            .await
+            .get(&key)
+            .map(|raw| serde_json::from_slice(raw).unwrap())
+            .unwrap_or_default();
+        // T isn't clone, so I can't use get
+        Ok(all.remove(id))
```

Comment:
Does this mean we for-realz remove it when we do this call? Or is it a copy

Reply:
It is just a copy as we are reading a HashMap from our in memory HashMap (yo dawg)